pkg/image/sif.go

```diff
@@ -78,6 +78,11 @@ func (f *sifFormat) initializer(img *Image, fi os.FileInfo) error {
 	}
 	defer fimg.UnloadContainer()
 
+	// It's a SIF, but an OCI-SIF.
+	if _, err := fimg.GetDescriptor(sif.WithDataType(sif.DataOCIRootIndex)); err == nil {
+		return debugErrorf("image is an OCI-SIF")
```

Comment:
Am I correct in understanding that this is a circumstance that shouldn't arise (i.e., there is code elsewhere that should make sure we don't reach this point if we're dealing with an OCI-SIF image)? And if so, wouldn't it be more accurate to characterize this circumstance as an "internal error" than a "debug error"...?

Reply:
No - this error is expected to be encountered. When Singularity's native runtime related code is provided an image file, it uses pkg/image to try and identify/load the image... and this attempts to call the initializer of each pkg/image type in turn. We can see this in the debug trace, for when I try to run an `oci-sif` with the native runtime:

```
$ singularity -d run  alpine_latest.sif 
DEBUG   [U=1001,P=1731880] persistentPreRun()            Singularity version: 3.11.0+462-g64fd3bd95
DEBUG   [U=1001,P=1731880] persistentPreRun()            Parsing configuration file /usr/local/etc/singularity/singularity.conf
DEBUG   [U=1001,P=1731880] maybeReExec()                 Checking whether to re-exec
DEBUG   [U=1001,P=1731880] handleConfDir()               /home/dtrudg-sylabs/.singularity already exists. Not creating.
DEBUG   [U=1001,P=1731880] handleRemoteConf()            Ensuring file permission of 0600 on /home/dtrudg-sylabs/.singularity/remote.yaml
DEBUG   [U=1001,P=1731880] launchContainer()             Using native runtime launcher.
DEBUG   [U=1001,P=1731880] setUmask()                    Saving umask 0022 for propagation into container
DEBUG   [U=1001,P=1731880] Init()                        Image format detection
DEBUG   [U=1001,P=1731880] Init()                        Check for sandbox image format
DEBUG   [U=1001,P=1731880] Init()                        sandbox format initializer returned: not a directory image
DEBUG   [U=1001,P=1731880] Init()                        Check for sif image format
DEBUG   [U=1001,P=1731880] Init()                        sif format initializer returned: image is an OCI-SIF
DEBUG   [U=1001,P=1731880] Init()                        Check for ocisif image format
DEBUG   [U=1001,P=1731880] Init()                        ocisif image format detected
FATAL   [U=1001,P=1731880] Exec()                        While checking image: native runtime does not support OCI-SIF images, use --oci mode
``` 

Note how it checks for a sandbox/dir first... then a SIF (non-OCI). Both of those initializers have to return a `debugError` to allow fall-through to trying the next initializer.

The initializer for ocsif succeeds... so the returned `img.Type` is `OCISIF`, and the rejection of execution is based on that.

The intent of the PR is really to modify `pkg/image` so it recognises `OCISIF` as a separate image type, and doesn't identify it as a SIF.